operator-common/src/main/java/io/strimzi/operator/common/Operator.java

```diff
@@ -94,9 +99,60 @@ default void reconcileThese(String trigger, Set<NamespaceAndName> desiredNames,
         return Optional.empty();
     }
 
-    Counter getPeriodicReconciliationsCounter();
+    Counter getPeriodicReconciliationsCounter(String namespace);
 
-    AtomicInteger getResourceCounter();
+    AtomicInteger getResourceCounter(String namespace);
 
-    AtomicInteger getPausedResourceCounter();
+    AtomicInteger getPausedResourceCounter(String namespace);
+
+    static Counter getCounter(String namespace, String kind, MetricsProvider metrics, Labels selectorLabels, Map<String, Counter> counterMap, String metricName, String metricHelp) {
+        String selectorValue = selectorLabels != null ? selectorLabels.toSelectorString() : "";
+        Tags metricTags = null;
+        if (namespace.equals("*")) {
+            metricTags = Tags.of(Tag.of("kind", kind), Tag.of("selector", selectorValue));
+        } else {
+            metricTags = Tags.of(Tag.of("kind", kind), Tag.of("namespace", namespace), Tag.of("selector", selectorValue));
+        }
+
+        Counter counter = counterMap.get(namespace + "/" + kind);
+        if (counter == null) {
+            counter = metrics.counter(metricName, metricHelp, metricTags);
+            counterMap.put(namespace + "/" + kind, counter);
+        }
```

Comment:
Use `computeIfAbsent` rather than separate `get` & `put`. With the ConcurrentHashMap that will give you atomicity which you need if you're not going to miscount things when different threads can access the map.